src/abstractdataframe/abstractdataframe.jl

```diff
@@ -189,6 +189,9 @@ rename(df, :i => :A, :x => :X)
 rename(df, [:i => :A, :x => :X])
 rename(df, Dict(:i => :A, :x => :X))
 rename(x -> Symbol(uppercase(string(x))), df)
+rename(df) do x
+    x |> string |> lowercase |> Symbol
```

Comment:
I guess you wanted `uppercase` not `lowercase` here. Right?

Reply:
Also I'd rather not use `|>` here to keeps things simple for newcomers.

Reply:
I think if noone shows how to use `|>`, newcomers won't understand how to use it.
This example being just after

    rename(x -> Symbol(uppercase(string(x))), df)

I think it helps to understand both `|>` and `do` block